### ISSUE TITLE
hunter parallel build of dependencies (make -jN)

### ISSUE THREAD

**Issue by headupinclouds** [ruslo/hunter#71]
Is there a mechanism to support parallel builds of hunter dependencies?

**Comment by ruslo** [comment]
Sure, I think it's quite easy but require careful testing since different tools use different flags. As far as I know:
* Makefile-like generators using `-jN`
* Xcode generator (xcodebuild tool) use `-jobs N`

Also I've made an option [--job](https://github.com/ruslo/polly/commit/724b68b1713173a205d120872c6b002406539666) in `build.py` script for Makefile and Xcode few days ago.

**Comment by ruslo** [comment]
Proposal:
* Introduce new variable `HUNTER_JOBS_NUMBER`
* Init this variable to NUMBER_OF_PHYSICAL_CORES (+1, +2, -1?) from [cmake_host_system_information](http://www.cmake.org/cmake/help/v3.2/command/cmake_host_system_information.html)

We can get some hints from this post also: http://www.kitware.com/blog/home/post/434

**Comment by headupinclouds** [comment]
I'll check out the --job option you added to see what that is doing.  (I'm guessing this doesn't run parallel builds of dependencies).  

FWIW, I recently found and added the following macro (source shown in comments below) for this purpose.  It sounds like you are already proposing something more flexible.  

```
macro(PARALLEL_MAKE)

# Macro from here: http://stackoverflow.com/questions/10688549/cmake-and-parallel-build
# Enable via CL i.e., -DMULTITHREADED_BUILD=8

#Add some multithreaded build support
MARK_AS_ADVANCED(MULTITHREADED_BUILD)
if(MULTITHREADED_BUILD)
    if(${CMAKE_GENERATOR} MATCHES "Unix Makefiles")
            message(STATUS ${CMAKE_BUILD_TOOL})
            set(CMAKE_MAKE_PROGRAM "${CMAKE_MAKE_PROGRAM} -j${MULTITHREADED_BUILD}")
            message(STATUS "Added arguments to CMAKE_BUILD_TOOL: ${CMAKE_MAKE_PROGRAM}")
    elseif(MSVC)
      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
      message(STATUS "Added parallel build arguments to CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
    endif()
endif()

endmacro(PARALLEL_MAKE)
```

And include that in my top level CMakeLists.txt
```
include(MacroParallelMake)
```

**Comment by ruslo** [comment]
> I'm guessing this doesn't run parallel builds of dependencies

Yep, just current build.

> I recently found and added the following macro

This can be applied via internal toolchain for all third party projects if MSVC detected:
```
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
```

This is same as `cmake --build . -- -jN` can be applied in `hunter_download` if Makefile detected:
```
set(CMAKE_MAKE_PROGRAM "${CMAKE_MAKE_PROGRAM} -j${MULTITHREADED_BUILD}")
```

**Comment by ruslo** [comment]
> I recently found and added the following macro
> ...
> set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")

Applied in [build.py](https://github.com/ruslo/polly/commit/807d47160c501829f4242efca73eb4cf4360ecac) script (branch `develop`)

**Comment by ruslo** [comment]
Implemented since: [v0.8.13](https://github.com/ruslo/hunter/releases/tag/v0.8.13)
Documentation: https://github.com/ruslo/hunter/wiki/usr.variables#hunter_jobs_number
